src/backend/aspen/api/views/phylo_runs.py

```diff
@@ -269,3 +271,28 @@ async def delete_run(
     await db.delete(item)
     await db.commit()
     return PhyloRunDeleteResponse(id=item_db_id)
+
+
+@router.put("/{item_id}")
+async def update_phylo_tree_and_run(
+    item_id: int,
+    phylo_run_update_request: PhyloRunUpdateRequest,
+    db: AsyncSession = Depends(get_db),
+    user: User = Depends(get_auth_user),
+):
+    # get phylo_run and check that user has permission to update PhyloRun/PhyloTree
+    phylo_run = await get_editable_phylo_run_by_id(db, item_id, user)
+
+    async with db as session:
+        # update phylorun name
+        phylo_run.name = phylo_run_update_request.name
+
+        # if there are any associated PhyloTrees update those names as well:
+        if phylo_run.outputs:
+            # there should only be one phylotree associated with the phylorun
+            phylo_tree = phylo_run.outputs[0]
+            phylo_tree.name = phylo_run_update_request.name
+
+        await session.commit()
+
+    return PhyloRunUpdateResponse(id=phylo_run.id)
```

Comment:
It's more common to return the same response body for an `update` request as a `get` request, but if the UI won't use the extra data 🤷‍♀️ I suppose this is a nit.

Reply:
a little confused, so should i rename `PhyloRunUpdateResponse` to be more generic (like `PhyloRunResponse`)? Or should i use `PhyloRunsListResponse`?

Reply:
It should probably return a `PhyloRunResponse`!